share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,660 @@
+```
+def incfirst fullstring chrstr = explode fullstring | tail | prepend chrstr | cat
+```
+
+* Function reorder takes a string , split the string and then reverses the splitted string and then concantenates both (Ex: hangover -> overhang)
+```
+def reorder fullstring = 
+  def Pair firststr laststr = 
+    explode fullstring
+    | splitAt (strlen fullstring / 2) 
+  cat (laststr ++ firststr)                         
+```
+
+* Function dubmid takes a string, duplicates the middle character and gives out the string (Ex: hapy -> happy))
+```
+def dubmid fullstring =
+  def Pair firststr laststr = 
+    explode fullstring
+    | splitAt (strlen fullstring / 2)
+  def dupli = laststr | take 1
+  cat (firststr ++ dupli ++ laststr)
+```
+
+* Takes a string and then removes the last character in a string
+```
+wake -x '"been"| explode | take 3| cat'
+```
+
+* Replaces the first charcter with the character "c"
+```
+wake -x '"bad"| explode | drop 1| prepend "c" | cat' 
+```
+### Tuple
+A tuple is collection of items of different types. `get`,`set` and `edit` are the three inbuilt methods that can be used with the tuples.
+* Below example shows the declaration of a tuple. 
+```
+tuple Collection =
+  global IntValue : Integer
+  global DoubleValue : Double
+  global StringValue : String
+```
+
+`set<tuple name><field Name>` 
+`set` operation sets(writes) value to the tuple field
+`get<tuple name><field Name>` 
+`get` operation gets(reads) value to the tuple field
+`edit<tuple name><field Name>` 
+`edit` operation moidfies value to the tuple field
+
+### List
+A List is a sequence of items of same type. Empty list returns `Nil`. `head` returns the first element in the list. `tail` returns the rest of the list without the `head`. 
+* Example shows how to prepend the list of integers with the given value
+```
+def d = 1,2,3,Nil
+def listprepend data1 = prepend data1 d
+```
+
+* Example creates a list of given elements
+```
+def listcreate xx = xx,xx,xx,Nil
+```
+* Try out these examples
+```
+wake -x '"south" | explode | take 1| cat'
+wake -x '"north"| explode | tail | take 1 | cat'
+wake -x '"east" | explode| reverse | take 1 |cat'
+wake -x '"west" | explode | reverse | tail |take 1| cat'
+```
+
+* Below are the set of examples which shows how to extract specific characters from a string
+```
+def first s = s | explode | take 1| cat
+def second s = s | explode | tail | take 1 | cat
+def third s = s | explode| tail | tail | take 1 |cat
+def fourth s =  s | explode | tail | tail | tail | take 1| cat
+def last s = s| explode | reverse |take 1 | cat
+```
+
+* Try out these different set of examples and capture the result
+```
+def roll s = "{fourth s}{first s}{second s}{third s}"
+def exch s = "{second s}{first s}{third s}{fourth s}"
+```
+
+* Below example shows how to declare a list of strings and apply map function to it. Map is a predefined function that maps to the every element in the list
```

Comment:
I thik you should add `map` to your list of important predefined functions, or explain each function in this way with an example. Mixing them up with some in the list and some only introduced in examples is confusing

Reply:
Done